### PR TITLE
remove autounify, fix langversion

### LINKED ISSUE #4386: dotnet migrate: buildOptions/languageVersion is not migrated properly
## Steps to reproduce

1. `dotnet new`
1. Add `"languageVersion": "csharp5"` to the `"buildOptions"` node.
1. `dotnet migrate`
1. `dotnet restore3`
1. `dotnet build3`

## Expected  behavior

The build should succeed. The .csproj should contain `<LangVersion>5</LangVersion>`.

## Actual behavior

The build fails:

```
CSC : error CS1617: Invalid option 'csharp5' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6. [C:\Users\joelv\Desktop\migrate\langver\langver.csproj]
```

## Workaround

Edit the .csproj `<LangVersion>csharp5</LangVersion>` to `<LangVersion>5</LangVersion>`.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-preview3-003830)

Product Information:
 Version:            1.0.0-preview3-003830
 Commit SHA-1 hash:  90a51bdd00

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
```

### PR DESCRIPTION
**Pull request by brthor**
Fixes #4386 (@joelverhagen)
Fixes auto unify being set in migration (@nguerrera)

/cc @livarcocc @eerhardt

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs [hunk 1]
```diff
@@ -355,5 +355,15 @@ private IncludeContext GetCopyToOutputIncludeContext(CommonCompilerOptions compi
                     null,
                     ProjectFilesCollection.DefaultPublishExcludePatterns);
         }
+
+        private string FormatLanguageVersion(string langVersion)
+        {
+            if (langVersion.StartsWith("csharp"))
```
Comment (joelverhagen) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: Are there other languages to consider here?

Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: Migration only migrates to csharp, there shouldn't be any cases where we're migrating a different value unless you can think of something.

Comment (joelverhagen) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: What about case sensitivity? If `CSHARP5` is in the project.json, does `dotnet build` still do the right thing?

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs [hunk 2]
```diff
@@ -355,5 +355,15 @@ private IncludeContext GetCopyToOutputIncludeContext(CommonCompilerOptions compi
...
+            {
+                return langVersion.Replace("csharp", "");
```
Comment (joelverhagen) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: and if you want replace (not sure why) nit: string.Empty

Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: fixed

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs (hunk 1 above)
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs]: good catch here

**Comment by brthor** [comment]
Also added a fix for the target framework property on project refs.

**Comment by brthor** [comment]
test Windows_NT x64 Release Build

**Pull request merged**